backend/accounting-service/src/main/kotlin/services/projects/v2/ProjectService.kt

```diff
@@ -1245,6 +1249,267 @@ class ProjectService(
         }
     }
 
+    suspend fun createInviteLink(actorAndProject: ActorAndProject, ctx: DBContext = db) {
+        val project = actorAndProject.requireProject()
+        val token = UUID.randomUUID().toString()
+
+        ctx.withSession { session ->
+            requireAdmin(actorAndProject.actor, listOf(project), session)
+
+            val count = session.sendPreparedStatement(
+                {
+                    setParameter("project", project)
+                },
+                """
+                    select * from project.invite_links where project_id = :project and now() < expires
+                """
+            ).rows.size
+
+            if (count >= 10) {
+                throw RPCException(
+                    "Unable to create more invitation links for this project",
+                    HttpStatusCode.BadRequest
+                )
+            }
+
+            val success = session.sendPreparedStatement(
+                {
+                    setParameter("project", project)
+                    setParameter("token", token)
+                },
+                """
+                    insert into project.invite_links (project_id, token, expires) values
+                        (:project, :token, now() + '30 days')
+                """
+            ).rowsAffected > 0
+
+            if (!success) {
+                throw RPCException(
+                    "Unable to create invitation link",
+                    HttpStatusCode.BadRequest
+                )
+            }
+        }
+    }
+
+    suspend fun browseInviteLinks(actorAndProject: ActorAndProject, ctx: DBContext = db): PageV2<ProjectInviteLink> {
+        val project = actorAndProject.requireProject()
+
+        val result = ctx.withSession { session ->
+            requireAdmin(actorAndProject.actor, listOf(project), session)
+
+            session.sendPreparedStatement(
+                {
+                    setParameter("project", project)
+                },
+                """
+                    select * from project.invite_links
```

Comment:
Would probably be better if this selects the required columns instead of selecting everything.